protocol-units/bridge/service/tests/relayer.rs

```diff
@@ -0,0 +1,347 @@
+
+		Ok(())
+	}
+	async fn get_bridge_transfer_details_with_nonce(
+		&mut self,
+		nonce: Nonce,
+	) -> BridgeContractResult<Option<BridgeTransferInitiatedDetails<MockAddress>>> {
+		todo!()
+	}
+	async fn is_bridge_transfer_completed(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId,
+	) -> BridgeContractResult<bool> {
+		todo!()
+	}
+}
+
+pub struct MockMonitoring {
+	listener: UnboundedReceiver<BridgeContractResult<BridgeContractEvent<MockAddress>>>,
+}
+
+impl MockMonitoring {
+	pub fn build(
+		listener: UnboundedReceiver<BridgeContractResult<BridgeContractEvent<MockAddress>>>,
+	) -> Self {
+		MockMonitoring { listener }
+	}
+}
+
+impl BridgeContractMonitoring for MockMonitoring {
+	type Address = MockAddress;
+}
+
+impl Stream for MockMonitoring {
+	type Item = BridgeContractResult<BridgeContractEvent<MockAddress>>;
+
+	fn poll_next(self: Pin<&mut Self>, cx: &mut std::task::Context) -> Poll<Option<Self::Item>> {
+		let this = self.get_mut();
+		this.listener.poll_next_unpin(cx)
+	}
+}
+
+async fn initiate_bridge_transfer(
+	initiator: MockAddress,
+	recipient: MockAddress,
+	amount: Amount,
+	nonce: Nonce,
+	sender: &mut UnboundedSender<BridgeContractResult<BridgeContractEvent<MockAddress>>>,
+) -> BridgeTransferId {
+	let bridge_transfer_id =
+		calculated_transfer_bridfe_id(&initiator.0, &recipient.0, amount, nonce);
+	let details = BridgeTransferInitiatedDetails {
+		bridge_transfer_id,
+		initiator: BridgeAddress(initiator.clone()),
+		recipient: BridgeAddress(recipient.clone().0),
+		nonce,
+		amount,
+	};
+	let event = BridgeContractEvent::Initiated(details);
+
+	sender.send(Ok(event)).await.unwrap();
+	bridge_transfer_id
+}
+
+#[tokio::test]
+async fn test_relayer_logic() -> Result<(), anyhow::Error> {
+	use tracing_subscriber::EnvFilter;
+	tracing_subscriber::fmt()
+		.with_env_filter(
+			EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
+		)
+		.init();
+
+	let l1_initiator_address = MockAddress(vec![11]);
+	let l1_recipient_address = MockAddress(vec![12]);
+
+	let l2_initiator_address = MockAddress(vec![21]);
+	let l2_recipient_address = MockAddress(vec![22]);
+
+	let (mut l1_sender, l1_listener) = futures::channel::mpsc::unbounded::<
+		BridgeContractResult<BridgeContractEvent<MockAddress>>,
+	>();
+	let (l1_relayer_client, mut l1_notifier) = RelayerMockClient::build(0, l1_sender.clone());
+	let l1_monitor = MockMonitoring::build(l1_listener);
+
+	let (mut l2_sender, l2_listener) = futures::channel::mpsc::unbounded::<
+		BridgeContractResult<BridgeContractEvent<MockAddress>>,
+	>();
+	let (l2_relayer_client, mut l2_notifier) = RelayerMockClient::build(0, l2_sender.clone());
+	let l2_monitor = MockMonitoring::build(l2_listener);
+
+	let (l1_health_tx, _l1_health_rx) = tokio::sync::mpsc::channel(10);
+	let (l2_health_tx, _l2_health_rx) = tokio::sync::mpsc::channel(10);
+	let (_health_tx, health_rx) = tokio::sync::mpsc::channel(10);
+	let _loop_jh = tokio::spawn(async move {
```

Comment:
Can we name this var to something else, not sure what jh refers to. Or just use `_` as its unused anyway

Reply:
I put jh for joinhandle. At first I thought I need to use it, but in fact no. I'll rename it with _.

Reply:
done